### ISSUE TITLE
How to play nice with hook id: no-commit-to-branch?

### ISSUE THREAD

**Issue by elgalu** [dmerejkowsky/tbump#106]
Hi, thank you so much for this tool!. Question:

When I'm not in the main/master branch I get `Error: Current branch (release) does not track anything. Cannot push.`

When I'm in the main/master branch it conflicts with [hook id: no-commit-to-branch](https://github.com/pre-commit/pre-commit-hooks#no-commit-to-branch) failing with `Don't commit to branch....`

Do you recommend to disable the `no-commit-to-branch` hook? or is there a workflow for tbump to work from a release branch?

**Comment by dmerejkowsky** [comment]
> Hi, thank you so much for this tool!

You're welcome :)

To answer your question -whether or not you should activate `no-commit-to-branch`really depends on your workflow. Let me elaborate.

The first workflow we can think of is to run `tbump` directly from the main development branch. (The `does not track anything` check is here to make sure the tag and the commit that bumps the version number are indeed pushed to the main development branch).

This workflow also works in case of "maintenance releases". For instance, after having published version 3.1 and 4.0, you may create a `release-3.x` branch and run `tbump 3.2` from it, creating the commit to bump to 3.2 and pushing it to the `release-3.x` branch.

So that's the main workflow I had in mind when writing `tbump` - but other are possible.

For instance, the Tanker projects (where I use to work), use this technique to make releases:

* Push a tag
* Let GitLab CI run `tbump --only-path <new-version>`
* Generate artifacts and upload them from GitLab CI

In conclusion, you can either:
* disable the `no-commit-to-branch` hook and use `tbump` directly
* keep the `no-commit-to-branch` hook and use `tbump --only-patch` to make releases from your CI/CD pipelines
* ... or anything else :)

Hope this helps! See `tbump --help` for all the ways you can customize tbump's behavior.

**Comment by elgalu** [comment]
thanks!
